sql/executor.go

```diff
@@ -165,28 +165,47 @@ func (e Executor) execStmt(stmt parser.Statement, params parameters, planMaker *
 				} else {
 					switch vt := val.(type) {
 					case parser.DBool:
-						row.Values = append(row.Values, driver.Datum{BoolVal: (*bool)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_BoolVal{BoolVal: bool(vt)},
```

Comment:
Oh, interesting. I'll have to correct my previous comment about an extra allocation occurring. `BoolVal` was previously a pointer, but now it is a value and `Datum_BoolVal` is a pointer. So this is the same number of allocations.